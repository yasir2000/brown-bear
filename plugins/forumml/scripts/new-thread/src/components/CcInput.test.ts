/*
 * Copyright (c) Enalean, 2021 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import CcInput from "./CcInput.vue";
import Vue from "vue";

describe("CcInput", () => {
    it("Displays the input field to enter a cc", () => {
        const wrapper = shallowMount(CcInput, {
            propsData: {
                cc: "toto@example.com",
                index: 123,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("Asks to parent component to remove the cc if user clicks on delete button", async () => {
        const wrapper = shallowMount(CcInput, {
            propsData: {
                cc: "toto@example.com",
                index: 123,
            },
        });

        const button = wrapper.find('[data-test="delete-cc"]');
        button.trigger("click");

        await Vue.nextTick();

        expect(wrapper.emitted().removeCc).toBeTruthy();
    });
});
