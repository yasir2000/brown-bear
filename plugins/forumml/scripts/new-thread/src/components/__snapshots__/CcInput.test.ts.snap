// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CcInput Displays the input field to enter a cc 1`] = `
<div
  class="tlp-form-element tlp-form-element-append forumml-post-new-thread-cc-component"
>
  <input
    class="tlp-input"
    id="forumml-cc-123"
    name="ccs[]"
    placeholder="jdoe@example.com"
    required="required"
    type="email"
  />
   
  <button
    class="tlp-append tlp-button-primary tlp-button-outline"
    data-test="delete-cc"
    type="button"
  >
    <i
      aria-hidden="true"
      class="fas fa-trash"
    />
  </button>
</div>
`;
